pokemongo_bot/cell_workers/pokemon_catch_worker.py

```diff
@@ -23,30 +23,52 @@ def __init__(self, pokemon, bot):
 
     def work(self):
         encounter_id = self.pokemon['encounter_id']
-        spawnpoint_id = self.pokemon['spawnpoint_id']
         player_latitude = self.pokemon['latitude']
         player_longitude = self.pokemon['longitude']
-        self.api.encounter(encounter_id=encounter_id, spawnpoint_id=spawnpoint_id,
-                           player_latitude=player_latitude, player_longitude=player_longitude)
+
+        if 'spawn_point_id' not in self.pokemon and 'fort_id' not in self.pokemon:
+            logger.log("Please update pgoapi using pip install --upgrade -r requirements.txt", 'red')
+            exit(0)
+
+        if 'spawn_point_id' in self.pokemon:
+            spawn_point_id = self.pokemon['spawn_point_id']
+            spawn_point_guid = spawn_point_id
+            response_key = 'ENCOUNTER'
+            response_status_key = 'status'
+            self.api.encounter(encounter_id=encounter_id, spawn_point_id=spawn_point_id,
+                               player_latitude=player_latitude, player_longitude=player_longitude)
+        else:
+            fort_id = self.pokemon['fort_id']
+            spawn_point_guid = fort_id
+            response_key = 'DISK_ENCOUNTER'
+            response_status_key = 'result'
+            self.api.disk_encounter(encounter_id=encounter_id, fort_id=fort_id,
+                                    player_latitude=player_latitude, player_longitude=player_longitude)
```

Comment:
Please, extract these verification into a private method. We're trying to avoid huge methods.